content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "GSoC 2022 Final Report: Improving Supply Chain Security"
+date: 2022-11-08
+draft: false
+weight: 100
+description: >
+  Going through my GSoC journey with Jenkins-X, this works as a documentation for our work to improve supply chain security of Jenkins X
+categories: [blog]
+keywords: [Community, GSoC, 2022, supply chain security, Report]
+slug: "gsoc-2022"
+aliases: []
+author: Osama Magdy
+---
+## Project Description
+
+  Due to the massive increase in open source projects usage in different applications and scales, it became a must to secure every project's supply chain of used packages and dependencies. Supply Chain Security is a rising concern in the current era and it's still under research and investigations. As part of making Jenkins-X an end-to-end solution for CI/CD I worked on both securing our own components and enabling our users to use these features during using our CI/CD for build and release steps.
+
+## Work Done
+
+  The work done so far was going through three main sections
+  1. Integrating with Tekton Chains to sign TaskRuns and PipelineRuns
+  2. Software Bill of Materials (SBOM)
+  3. Signing Jenkins-X artifacts
+
+### Integrating with Tekton Chains to sign TaskRuns and PipelineRuns
+
+  ***Description:***
+
+    As Jenkins X uses tekton as its pipeline execution engine,` TaskRun`s and ` PipelineRun`s are considered the key components of Jenkins X pipeline `activities ` and  `steps `
```

Comment:
Something funky with the formatting, just remove the plural s from TaskRun and PipelineRUn

Reply:
Done